tla/etcdraft.tla

```diff
@@ -686,43 +796,49 @@ NextAppendEntriesResponse == \E m \in DOMAIN messages : m.mtype = AppendEntriesR
 
 \* The network duplicates a message
 \* @type: MSG => Bool;
-DuplicateMessage(m) ==
+DuplicateMessage(m) == SpecAction("DuplicateMessage", <<m>>, LAMBDA act:
     /\ m \in DOMAIN messages
     /\ messages' = WithMessage(m, messages)
-    /\ UNCHANGED <<pendingMessages, serverVars, candidateVars, leaderVars, logVars, configVars, durableState>>
+    /\ UNCHANGED <<pendingMessages, serverVars, candidateVars, leaderVars, logVars, configVars, readyVars>>
+)
 
 \* The network drops a message
 \* @type: MSG => Bool;
-DropMessage(m) ==
+DropMessage(m) == SpecAction("DropMessage", <<m>>, LAMBDA act:
     \* Do not drop loopback messages
     \* /\ m.msource /= m.mdest
     /\ Discard(m)
-    /\ UNCHANGED <<pendingMessages, serverVars, candidateVars, leaderVars, logVars, configVars, durableState>>
+    /\ UNCHANGED <<pendingMessages, serverVars, candidateVars, leaderVars, logVars, configVars, readyVars>>
+)
 
 ----
 
 \* Defines how the variables may transition.
-NextAsync == 
+Next == 
     \/ \E i,j \in Server : RequestVote(i, j)
     \/ \E i \in Server : BecomeLeader(i)
     \/ \E i \in Server: ClientRequest(i, 0)
     \/ \E i \in Server : AdvanceCommitIndex(i)
-    \/ \E i,j \in Server : \E b,e \in matchIndex[i][j]+1..Len(log[i])+1 : AppendEntries(i, j, <<b,e>>)
+    \/ \E i,j \in Server : 
```

Comment:
Typically, `Next` in large specifications only quantifies over constants to clearly indicate which variables are changed by sub-actions like `AppendEntries`.  Therefore, it would be better to move the quantification over `matchIndex` and `log` out of `Next` and into the definition of `AppendEntries`. This change will also have the side effect of making TLC display the names of sub-actions, such as `AppendEntries`, in counterexamples.

Reply:
I'll fix it in the new commit.

Reply:
Fixed in the new commit.